src/api/user/test/e2e/user.test.js

```diff
@@ -0,0 +1,360 @@
+const request = require('supertest');
+const firebaseTesting = require('@firebase/rules-unit-testing');
+
+const { app } = require('../../index');
+const User = require('../../src/models/user');
+
+// Utility functions
+const clearData = async () => {
+  await firebaseTesting.clearFirestoreData({ projectId: 'telescope' });
+};
+
+const getUser = (id) => request(app).get(`/${id}`);
+
+const getUsers = () => request(app).get('/');
+
+const postUser1 = async () => {
```

Comment:
Instead of using global data like this, let's create a function you call when you need it:

```js
const createUser = async (user = {}) => {
  const defaultUser = {
    // Use a unique id number for each user
    id: Date.now(),
    firstName: 'Galileo',
    lastName: 'Galilei',
    displayName: 'Galileo Galilei',
    isAdmin: true,
    isFlagged: true,
    feeds: ['https://dev.to/feed/galileogalilei'],
    github: {
      username: 'galileogalilei',
      avatarUrl:
        'https://avatars.githubusercontent.com/u/7242003?s=460&u=733c50a2f50ba297ed30f6b5921a511c2f43bfee&v=4',
    },
  };
  // If the user sends use any user properties, override the default values.  If not, use default as is.
  const user = new User({...defaultUser, ...user});
  const response = wait request(app).post('/').set('Content-Type', 'application/json').send(user);
  // Return both the user object and the response, so we can compare the two.
  return { user, response };
}l
```

Now in your tests below, any time you need a user inserted, you can do this:

```js
// Create a default user and insert
const { user, response } = await createUser();
// The user's id is unique, but we can get it like this:
const id = user.id;
```

If we need to test something special, we can override a value:

```js
// Create a default user but force the id to be 10001
const { user, response } = await createUser({ id: 10001 });
```